esphome/components/ssd1306_base/ssd1306_base.cpp

```diff
@@ -69,25 +97,37 @@ void SSD1306::setup() {
     case SH1106_MODEL_128_64:
     case SSD1306_MODEL_64_48:
     case SH1106_MODEL_64_48:
+    case SSD1305_MODEL_128_32:
+    case SSD1305_MODEL_128_64:
       this->command(0x12);
       break;
   }
 
+  // Pre-charge period (0xD9)
   this->command(SSD1306_COMMAND_SET_PRE_CHARGE);
   if (this->external_vcc_)
     this->command(0x22);
   else
     this->command(0xF1);
 
+  // Set V_COM (0xDB)
   this->command(SSD1306_COMMAND_SET_VCOM_DETECT);
   this->command(0x00);
 
+  // Display output follow RAM (0xA4)
   this->command(SSD1306_COMMAND_DISPLAY_ALL_ON_RESUME);
-  this->command(SSD1306_NORMAL_DISPLAY);
 
+  // Inverse display mode (0xA6, 0xA7)
+  this->command(SSD1306_COMMAND_NORMAL_DISPLAY | this->invert_);
+
+  // Disable scrolling mode (0x2E)
   this->command(SSD1306_COMMAND_DEACTIVATE_SCROLL);
 
-  set_brightness(this->brightness_);
+  // Contrast and brighrness
+  // SSD1306 does not have brightness setting
+  set_contrast(this->contrast_);
+  if (!this->is_ssd1305_())
```

Comment:
Is this supposed to be only for ssd1305?
```suggestion
  if (this->is_ssd1305_())
```

Reply:
Yes, a stupid mistake :-) Thanks for pointing out